test/unit/migrations/multiple_ldap_hostnames_migration_test.rb

```diff
@@ -0,0 +1,107 @@
+# encoding: utf-8
+
+#  Copyright (c) 2008-2017, Puzzle ITC GmbH. This file is part of
+#  Cryptopus and licensed under the Affero General Public License version 3 or later.
+#  See the COPYING file at the top-level directory or at
+#  https://github.com/puzzle/cryptopus.
+
+require 'test_helper'
+migration_file_name = Dir[Rails.root.join('db/migrate/20171010071016_multiple_ldap_hostnames.rb')].first
+require migration_file_name
+
+class MultipleLdapHostnamesMigrationTest < ActiveSupport::TestCase
+
+  before(:each) do
+    ldap_hostname_setting.destroy
+  end
+
+  describe '#up' do
+    test 'does not change type if hostname is an array' do
+      Setting.create!(key: 'ldap_hostname', value: ['yourdomain.com'], type: 'Setting::Host')
+
+      migration.up
+
+      assert_equal 'Setting::Host', ldap_hostname_setting.type
+      assert_equal ['yourdomain.com'], ldap_hostname_setting.value
+    end
+
+    test 'does not change type if hostname not exists' do
+      Setting.create!(key: 'ldap_hostname', value: nil, type: 'Setting::Text')
+
+      migration.up
+
+      assert_equal 'Setting::Text', ldap_hostname_setting.type
```

Comment:
after this migration, ldap_hostname_setting should be a Setting::Host in any cases since this is what this new feature requires